Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/CoinsConfig.hs

```diff
@@ -0,0 +1,96 @@
+module Domain.Action.Dashboard.Management.CoinsConfig
+  ( putCoinsConfigUpdate,
+    postCoinsConfigCreate,
+  )
+where
+
+import qualified API.Types.ProviderPlatform.Management.CoinsConfig as Common
+import qualified Data.Text as Text
+import qualified Domain.Types.Coins.CoinsConfig as DTCC
+import qualified Domain.Types.Merchant
+import Domain.Types.Translations (Translations (..))
+import Domain.Types.VehicleCategory (VehicleCategory (CAR))
+import qualified Environment
+import EulerHS.Prelude hiding (id)
+import Kernel.Types.APISuccess (APISuccess (Success))
+import qualified Kernel.Types.Beckn.Context
+import Kernel.Types.Error (GenericError (InvalidRequest))
+import qualified Kernel.Types.Id as ID
+import qualified Kernel.Utils.Common as UC
+import qualified Storage.CachedQueries.CoinsConfig as CQConfig
+import qualified Storage.Queries.Coins.CoinsConfig as QConfig
+import Storage.Queries.Translations (create)
+import Storage.Queries.TranslationsExtra (isTranslationExist)
+
+putCoinsConfigUpdate ::
+  ID.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  Common.UpdateReq ->
+  Environment.Flow Kernel.Types.APISuccess.APISuccess
+putCoinsConfigUpdate _merchantShortId _opCity updateReq@Common.UpdateReq {..} = do
+  coinsConfig <-
+    QConfig.findById (ID.cast entriesId)
+      >>= UC.fromMaybeM (InvalidRequest "Coins config does not exist")
+  _ <- do
+    QConfig.updateCoinEntries updateReq
+    let vehicleCategory = fromMaybe CAR coinsConfig.vehicleCategory
+    CQConfig.clearCache coinsConfig.eventName coinsConfig.eventFunction (ID.Id coinsConfig.merchantOptCityId) vehicleCategory
+  pure Success
+
+postCoinsConfigCreate ::
+  ID.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  Common.CreateCoinsConfigReq ->
+  Environment.Flow Kernel.Types.APISuccess.APISuccess
+postCoinsConfigCreate _merchantShortId _opCity req = do
+  uuid <- UC.generateGUIDText
+  (newCoinsConfig, eventMessageLs) <- case req of
+    Common.NewCoinsConfig (Common.NewCoinsConfigReq {..}) ->
+      pure
+        ( DTCC.CoinsConfig
+            { id = ID.Id uuid,
+              ..
+            },
+          eventMessages
+        )
+    Common.DuplicateCoinsConfig (Common.DuplicateCoinsConfigsReq {..}) -> do
+      coinsConfig <-
+        QConfig.findById (ID.cast entriesId)
+          >>= UC.fromMaybeM (InvalidRequest "Coins config does not exist")
+      pure
+        ( coinsConfig
+            { DTCC.id = ID.Id uuid,
+              DTCC.eventFunction = eventFunction
+            },
+          eventMessages
+        )
+  _ <- do
+    QConfig.createCoinEntries newCoinsConfig
+    CQConfig.clearCache
+      newCoinsConfig.eventName
+      newCoinsConfig.eventFunction
+      (ID.Id newCoinsConfig.merchantOptCityId)
+      (fromMaybe CAR newCoinsConfig.vehicleCategory)
```

Comment:
I think we don't need to clear cache if `vehicleCategory` is Nothing., because both redis keys
 contain `vehicleCategory`:

```
makeCoinConfigKey :: Text -> Id DMOC.MerchantOperatingCity -> DTV.VehicleCategory -> Text

makeCoinConfigOnEventAndFunctionKey :: Text -> DCT.DriverCoinsFunctionType -> Id DMOC.MerchantOperatingCity -> DTV.VehicleCategory -> Text
```